### ISSUE TITLE
[Trust Model] validation of Metadata, jwks and not repudiable trust relationships

### ISSUE THREAD

**Issue by peppelinux** [vcstuff/draft-terbu-sd-jwt-vc#33]
In https://github.com/vcstuff/draft-terbu-sd-jwt-vc/blob/main/draft-terbu-sd-jwt-vc.md#jwt-issuer-metadata-jwt-issuer-metadata

My concerns rely about the semantic of this `/.well-known/jwt-issuer`, since every participant in a identity (openid) or authorization (Oauth) infrastructure is in fact a jwt issuer, even the relying party issues JWT (authz request, private_key_jwt ...). I would like to stress our brains to get a solution re-using what we already have or specialized other wel-known(s) endpoint in relation at the new roles in the wallet ecosystem

In addition to this "intellectual" challenge, I propose to do a PR by introducing `.well-known/openid-federation` for secure trust establishment mechanisms that cannot be repudiated over time.

In particular the role of the `trust_chain` JWS header as shown in the example below

````
{
 "alg": "ES256",
 "typ": "vc+sd-jwt",
 "trust_chain": [
   "fIIE3jCCA8agAwIBAgICAwEwT...TWVU",
   "aiIE+zCCBGSgAwIBAgICAQ0wD...09VZ",
   "fyIC5zCCAlACAQEwDQYJKoZIh...9ViH"
 ]
}
.
{
 "iss": "https://credential-issuer.example.com/",
 "iat": 1516239022,
 "exp": 1516247022,
 "type": "Identity",
 "_sd": [
   "UiuRGkTW7e_5UQauGeQRQdF8u3WYevS4Fs0IuB_DgYM",
   "tmPlXq0MID-oRXbUNHyoVZrc9Qkm8cwJTohVyOVlUgQ",
   "vTz0JI103v4k4pKIloT83Yzi33L1SdZlWBPmsfJBefk"
 ],
 "_sd_alg": "sha-256",
 "cnf": {
   "jwk": {
     "kty": "EC",
     "crv": "P-256",
     "x": "TCAER19Zvu3OHF4j4W4vfSVoHIP1ILilDls7vCeGemc",
     "y": "ZxjiWWbZMQGHVWKVQ4hbSIirsVfuecCE6t4jT9F2HZQ"
   }
 }
}
````
the benefits of the federation trust chain are summarized below

1. trust and protocol specific jwks are available in it
2. the jwks and the metadata are verifiable having just the public key of a single trusted third party (Trust Anchor)
3. the metadata of the issuer are subject of the policies of the trust chain, the trust chain may overload, add,remove, some elements from the final metadata by applying the policies contained in it. This allow us to remove the possibility that an entity may use a weak alg or defines unsupported scopes or any other property not allowed by the regulation implemented by Trust Anchor

This JWS header will be registered in the IANA registry and will be soon published in the OIDC Federation draft 29

I can do the draft of the PR and ask to you a revision as usual, while regarding the first point I'd suggest more discussion on that

**Comment by tlodderstedt** [comment]
The way the spec is written now, the .well-known/jwt-issuer location can be used by anyone issuing JWTs (as long as the issuer value is a HTTPS URL). I think that is ok. If you want to tighten the scope or change the name, we can talk about it. .well-known/jwt-issuer just makes the OpenID Connect approach for key management available to the world of verifiable credentials and people seem to like the idea. 

You are suggesting a completely different approach (along the lines of OpenID Federation). That's fine (and you know I like your passion about it) and implementers can use this mechanism with SD-JWT VCs (in the same way they can use OpenID Federation as alternative to the OpenID Connect Key resolution today). 

So what is the issue?

**Comment by peppelinux** [comment]
Yes, the scope of this instance should really be defined since it seems too much wide

This Issue wants place in the specs a clear statement, and non-normative example, about the key resolution following the approach of the trust relationships, then X509 certificate chain and OpenID Connect Federation Trust Chain

the requirement is that there are many cases where a decentralized infrastructure needs a trust delegation mechanism where the verifiable attestation of the trust relationship between the entities is represented as a "chain", made with linked claims, public keys and signature bindings 

If the defining of the trust relationship verifiable attestations is out of scope for this specs, I suggest to not propose the sole OAuth2 connect approach but giving more non-normative example linked to external refs

**Comment by tlodderstedt** [comment]
The .well-known/jwt-issuer is a simple mechanism for key resolution and key rotation. 

Trust management can be built on top. That's why it is out of scope.

**Comment by peppelinux** [comment]
My concerns is related to the missing requirements related to the trust model, that expose us in defining new specification following the hypes or the feedback taken along the road with the mechanism of the empathy or other experiences characterized by social exchanges, these are important but they may have distractions and distorsion due to enviromental conditions or individual inclinations.

since I believe that this is the right place to do the right things, I'd like to start over the requirements of the trust model and key attestation mechanisms as well

In this document me and other contributors are trying to define an agnostic approach in the definition of the trust model requirements and rules, given as granted the architectural requirements of decentralization

[The eID Wallet Trust Model Rules](https://docs.google.com/document/d/1Dk_8UmytCI4VhCx5VMnXmEzdXRvgJozGeq1GNHZOQik/)

I'm looking to non repudiation, decentralized historical registry of expired/revoked keys, selfconsistent VC/Attestation with verifiable keys, metadata and policies

that's why I propose a PR including .well-known/openid-federation as endpoint or the way we may define to integrate jwt-issuer with the federation trust chains, since these give the full coverage of the requirements

the prerequisite is then the cryptographic verifiability, so jwt-issuer response should or must be signed

then I've noted that in the text is written that only x509 and .well-known/jwt-issuer MUST be supported, this would exclude the current and the future specification that brings an higher level of assurance on the attestations. I'd remove that "only" :-)

**Comment by tlodderstedt** [comment]
Hi Giuseppe, it seems you are mixing up the scope of vc-sd-jwt and the high assurance profile. This is the repo of vc-sd-jwt. It won't define mandatory to implement requirements for key resolution & issuer identification/authentication. It just introduces a simple mechanism for this purpose. Mandatory to implement requirement are stated in the high assurance profile. 
PS: I don't see why .well-known/jwt-issuer must be signed as pre-requisite to use it with trust management mechanisms, including federation. .well-known/jwt-issuer is intended to be super simple. Application level encryption adds complexity (itself and through key management) and does not add security for an API based model. It makes sense in a mode where all data is carried in a chain structure, but that is a different story.

**Comment by peppelinux** [comment]
Yes, I've mixed up many things just for giving an harmonic view on the requirements and assumptions I have in mind

Considering how we can work together to improve high assurance profile, I've raised the issue of having in sd-jwt-vc even openid-federation and not x5c and jwt-issuer alone, as raised in several other issues (and here my excuse for the noise)

During the previous connect call there was the agreement that once Federation Draft 29 will be available, then we'll be able to include the Federation trust_chain JWS header parameter in all the OpenID4VC specs, so I'm looking forward to have this agreement also in sd-jwt-vc 

As you know, mine is not just a request but a manifestation of willingness to contribute actively

**Comment by tlodderstedt** [comment]
> During the previous connect call there was the agreement that once Federation Draft 29 will be available, then we'll be able to include the Federation trust_chain JWS header parameter in all the OpenID4VC specs, so I'm looking forward to have this agreement also in sd-jwt-vc

That's what the minutes say

<img width="576" alt="image" src="https://github.com/vcstuff/draft-terbu-sd-jwt-vc/assets/328358/8e0d7028-8fd6-435f-bdbc-c602f7a320b4">

Interestingly, the issue was closed without any further comments. 

You can certainly open an issue on this in the issue tracker and we will discuss this in the SIOP special calls (or the OID4VC calls) going forward.

**Comment by awoie** [comment]
I agree with @tlodderstedt that this issue relates to the OpenID4VC hi-assurance profile. This spec is a general-purpose Verifiable Credentials representation based on SD-JWTs.

Closing this issue. Please re-open if you disagree.

**Comment by peppelinux** [comment]
draft 29 has its changelog with the fixed issues by id number, the review process has been completed in the PR, then this latter is referenced in the Issue

https://bitbucket.org/openid/connect/issues/1921/trust_chain-as-jwt-header-claim

ok guys
